File: Week_04/assign_cookies/Solution.java
package assign_cookies;

import java.util.Arrays;

/**
 * 贪心法：优先用最小的饼干满足胃口最小的孩子
 * Just assign the cookies starting from the child with less greediness to maximize the number of happy children
 */
class Solution {
    public int findContentChildren(int[] g, int[] s) {
        Arrays.sort(g);
        Arrays.sort(s);
        int gi = 0, si = 0;
        while (gi < g.length && si < s.length) {
            if (g[gi] <= s[si])
                gi++;
            si++;
        }
        return gi;
    }
}

File: Week_04/assign_cookies/Solution.py
class Solution:
    def findContentChildren(self, g: List[int], s: List[int]) -> int:
        g.sort()
        s.sort()
        i, j = 0, 0
        while i < len(g) and j < len(s):
            if g[i] <= s[j]: #饼干小于等于孩子的胃口时 = 可以满足当前孩子
                i += 1 #轮到下一个孩子
            j += 1 #继续拿出下一个饼干，无论满不满足当前孩子
        return i
